src/solvers/primal_dual_semismooth_Newton.jl

```diff
@@ -0,0 +1,378 @@
+using SparseArrays
+
+@doc raw"""
+    primal_dual_semismooth_Newton(M, N, cost, x0, ξ0, m, n, prox_F, diff_prox_F, prox_G_dual, diff_prox_dual_G, linearized_operator, adjoint_linearized_operator)
+
+Perform the Primal-Dual Riemannian Semismooth Newton algorithm.
+
+Given a `cost` function $\mathcal E\colon\mathcal M \to \overline{ℝ}$ of the form
+```math
+\mathcal E(x) = F(x) + G( Λ(x) ),
+```
+where $F\colon\mathcal M \to \overline{ℝ}$, $G\colon\mathcal N \to \overline{ℝ}$,
+and $\Lambda\colon\mathcal M \to \mathcal N$. The remaining input parameters are
+
+* `x,ξ` primal and dual start points $x\in\mathcal M$ and $\xi\in T_n\mathcal N$
+* `m,n` base points on $\mathcal M$ and $\mathcal N$, respectively.
+* `linearized_forward_operator` the linearization $DΛ(⋅)[⋅]$ of the operator $Λ(⋅)$.
+* `adjoint_linearized_operator` the adjoint $DΛ^*$ of the linearized operator $DΛ(m)\colon T_{m}\mathcal M \to T_{Λ(m)}\mathcal N$
+* `prox_F, prox_G_Dual` the proximal maps of $F$ and $G^\ast_n$
+* `diff_prox_F, diff_prox_dual_G` the (Clarke Generalized) differentials of the proximal maps of $F$ and $G^\ast_n$
+
+For more details on the algorithm, see[^DiepeveenLellmann2021].
+
+# Optional Parameters
+
+* `primal_stepsize` – (`1/sqrt(8)`) proximal parameter of the primal prox
+* `Λ` (`missing`) the exact operator, that is required if `Λ(m)=n` does not hold;
+`missing` indicates, that the forward operator is exact.
+* `dual_stepsize` – (`1/sqrt(8)`) proximal parameter of the dual prox
+* `reg_param` – (`1e-5`) regularisation parameter for the Newton matrix
+Note that this changes the arguments the `forward_operator` will be called.
+* `stopping_criterion` – (`stopAtIteration(50)`) a [`StoppingCriterion`](@ref)
+* `update_primal_base` – (`missing`) function to update `m` (identity by default/missing)
+* `update_dual_base` – (`missing`) function to update `n` (identity by default/missing)
+* `retraction_method` – (`ExponentialRetraction()`) the rectraction to use
+* `inverse_retraction_method` - (`LogarithmicInverseRetraction()`) an inverse retraction to use.
```

Comment:
...you might have guessed by now, this needs to be `default_vector_transport_method(M)`.

Reply:
Changed